agent/command/git.go

```diff
@@ -870,15 +871,29 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		}
 	}
 
-	// Clone the project's modules.
+	// We use context.Background() instead of ctx here because
```

Comment:
Could you explain this a bit more? How does it ruin a honeycomb trace? Is this a side effect of using `x/sync/errgroup`, and instead maybe we should use the `sync.WaitGruop` approach from elsewhere?

Reply:
This under the hood just wraps wait group and catches errors, cancelling the provided context. This ensures that if one has failed before, it does not continue running cloning code despite erroring later.

Since it cancels the provided context too, honeycomb does not like that the context has been modified. Then the span on honeycomb is not correct for the rest of the span (since the context changed). It's not important to use our context here, but to use _some_ context it can latch on to. I could change the comment to better describe this if you think so!

Reply:
I'm maybe misunderstanding what you're saying, but if you make a child context from a parent context, and then cancel the child, it does not cancel the parent. (Parents cancel their children, but children do not cancel their parents.) Are you saying that honeycomb _also_ has a problem if you cancel child contexts?

Reply:
No, how errgroup is intended to be used it propogates errors to the parent passed in- which honeycomb tests for and stops putting in the span if so. I tested it using ctx instead of context.Background() and honeycomb's span stops after this function with only that change

Reply:
e.g. if you look at the honeycomb link I sent for the original parallel module + project (last link in the desc) you do not see all the span items honeycomb should have, it ends at clone module (while the others have more info). That one uses the same context throughout

Reply:
> No, how errgroup is intended to be used it propogates errors to the parent passed in

I disagree with this, though I might be missing something. I read the code as deriving a new context.

https://cs.opensource.google/go/x/sync/+/refs/tags/v0.6.0:errgroup/errgroup.go;l=48

And the docs explicitly call out the derived context as the thing that gets canceled.

> The derived Context is canceled the first time a function passed to Go returns a non-nil error or the first time Wait returns, whichever occurs first.

Reply:
Hm I read that too but honeycomb just doesn't continue the span if I use our context there. It was driving me crazy why but only when I used a new context did honeycomb continue the span. What do you think might be the case?

Reply:
From Slack conversations it looks like the Honeycomb links are a fraction of a second too short, so the traces are getting sent, but the link in the UI cuts off the last fraction of a second of spans.

Reply:
I did it with the same context again and honeycomb resulted in: https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen-agent/trace/xb4xs7i5zNf?fields%5B%5D=s_name&fields%5B%5D=s_serviceName&span=53c65e1ce7122c35